opensrp-app/src/main/java/org/smartregister/authorizer/P2PSyncAuthorizationService.java

```diff
@@ -0,0 +1,44 @@
+package org.smartregister.authorizer;
+
+import android.support.annotation.NonNull;
+
+import org.smartregister.AllConstants;
+import org.smartregister.p2p.authorizer.P2PAuthorizationService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by Ephraim Kigamba - ekigamba@ona.io on 22/03/2019
+ */
+
+public class P2PSyncAuthorizationService implements P2PAuthorizationService {
+
+    private Map<String, Object> authorizationDetails = new HashMap<>();
+
+    public P2PSyncAuthorizationService(@NonNull String teamId) {
+        authorizationDetails.put(AllConstants.PeerToPeer.KEY_TEAM_ID, teamId);
+    }
+
+    @Override
+    public void authorizeConnection(@NonNull Map<String, Object> peerDeviceMap, @NonNull final AuthorizationCallback authorizationCallback) {
+        getAuthorizationDetails(new OnAuthorizationDetailsProvidedCallback() {
+            @Override
+            public void onAuthorizationDetailsProvided(@NonNull Map<String, Object> map) {
+                Object peerDeviceTeamId = map.get(AllConstants.PeerToPeer.KEY_TEAM_ID);
```

Comment:
retrieve `peerDeviceTeamId` from `peerDeviceMap`. 

`map` is the `authorizationDetails` which the device details, so currently the device is compairing its team_id against its team_id